### ISSUE TITLE
Diagrams are not shown on print preview

### ISSUE THREAD

**Issue by slauriere** [xwikisas/application-diagram#185]
## Steps to reproduce

1. Create a diagram
2. Add a shape to it
3. Save and view the page
4. Check that the files "diagram.png" and "diagram.svg" are present as page attachments
5. Hit More Actions > Print Preview

## Actual result

* The diagram shows up for a few milliseconds and disappears instead of remaining visible

NB: there is no error in the JavaScript console

## Expected result

* The diagram is present on the page and remains visible without disappearing

## Environment

* Diagram Pro 1.15.5
* XWiki 13.0
* Firefox or Chrome

**Comment by lucaa** [comment]
@slauriere please add the "expected result" to this ticket (even if it may seem obvious) so that it's explicit what doesn't actually work. Also, detail a little point 4 from the steps to reproduce: the "check" means that at that point the 2 files are expected to be present as page attachments?

**Comment by slauriere** [comment]
@lucaa I've updated the ticket description indeed: "Expected result" section added + point 4 about the file check slightly reworded

**Comment by ane-gabriela** [comment]
Issue reproduces on XWiki 12.10.5 and 9.11, Chrome 89, Diagram Application (Pro) 1.16-rc-2

Diagram:
![DiagramView](https://user-images.githubusercontent.com/40691112/115578349-e17d1b00-a2cd-11eb-9bdd-1880d06e6dad.png)

More actions > Print Preview:
![PrintPreview](https://user-images.githubusercontent.com/40691112/115578407-f0fc6400-a2cd-11eb-9e8c-d116a94e571e.png)

**Comment by mflorea** [comment]
The print preview loads [``print.css``](https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-12.10.7/xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/print.css#L7) which forces this style:

```
#body, #xwikimaincontainer, #xwikimaincontainerinner, #xwikicontent, #mainContentArea {
  ...
  width: auto !important;
}
```

which makes the diagram container (child of ``#xwikicontent``) have 0 width. This in turn triggers this code from [``GraphViewer.js``](https://github.com/jgraph/drawio/blob/v14.6.13/src/main/webapp/js/diagramly/GraphViewer.js#L441):

```
if (this.checkVisibleState && container.offsetWidth == 0 && typeof MutObs !== 'undefined') {
  // Delayed rendering if inside hidden container and event available
  ...
```

Basically draw.io thinks the diagram container is hidden and thus delays (by default) the rendering until it becomes visible. Which doesn't happen in our case because our diagram container is visible but has 0 width. The easy fix is to set a minimum width on the diagram container.
